awx/main/tasks/facts.py

```diff
@@ -36,9 +37,8 @@ def start_fact_cache(hosts, destination, log_data, timeout=None, inventory_id=No
 
     if isinstance(hosts, QuerySet):
         hosts = hosts.iterator()
-
     last_filepath_written = None
-    for host in hosts:
+    for host in hosts_cached:
```

Comment:
The `start_fact_cache` still needs to get the host list from the database, which is `job.get_hosts_for_fact_cache()`, which is still passed in as `hosts` to this method still. It's only the finishing method that should defer to the cached hosts from the starting method... somehow.

Reply:
Thank you so much!! I've fixed that, this is very helpful!